From ff0679b8f844847057df4d8fd31e78dd0205c0a0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20K=C3=BCnzel?= <julius.kuenzel@kde.org>
Date: Sun, 22 Feb 2026 13:40:57 +0100
Subject: [PATCH] chore!: make dependency management explicite

Right now if WITHOUT_XY is OFF, but XY is not found, it behaves the same as if WITHOUT_XY was ON. This behavior ignores the user intent and makes it harder to get consistent and reproducible builds.

With this patch the user intent is honored:
- If WITHOUT_XY is ON frei0r will be build without filters depending on XY (same as before)
- If WITOUT_XY is OFF and XY is found, frei0r will be build with filters depending on XY (same as before)
- If WITOUT_XY is OFF and XY is not found, the build will fail in the CMake configure stage. The user has to either set WITOUT_XY=OFF if they don't want the dependency or install it.

Also for Cairo there was no option to control whether it should be used yet. This patch introduces WITHOUT_CAIRO similar to the already existing WITHOUT_GAVL and WITHOUT_OPENCV
---
 CMakeLists.txt | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9bb5b8cf..dd0e14ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,15 +11,18 @@ option (WITHOUT_OPENCV "Disable plugins dependent upon OpenCV" OFF)
 option (WITHOUT_FACERECOGNITION "Disable facedetect plugin to avoid protobuf conflicts" OFF)
 
 if (NOT WITHOUT_OPENCV)
-  find_package (OpenCV)
+  find_package (OpenCV REQUIRED)
 endif ()
 
-find_package (Cairo)
+option (WITHOUT_CAIRO "Disable plugins dependent upon gavl" OFF)
+if (NOT WITHOUT_CAIRO)
+  find_package (Cairo REQUIRED)
+endif ()
 
 include(FindPkgConfig)
 option (WITHOUT_GAVL "Disable plugins dependent upon gavl" OFF)
 if (PKG_CONFIG_FOUND AND NOT WITHOUT_GAVL)
-  pkg_check_modules(GAVL gavl)
+  pkg_check_modules(GAVL REQUIRED gavl)
 endif ()
 
 include_directories (AFTER include)
